### DISCUSSION TITLE
Path StackCount pushd Popd

### DISCUSSION THREAD

**Discussion by QNetSo** [JanDeDobbeleer/oh-my-posh#2300] [Troubleshoot]
I had this working with an earlier 3.x version of oh-my-posh (wished I could tell you the version) ... but it no longer works. my depth is always zero. I found this in the 2.x docs [adding-stack-count-to-a-custom-theme](https://github.com/JanDeDobbeleer/oh-my-posh2#adding-stack-count-to-a-custom-theme) but not sure how to get it working in the newer 3.x versions.

this is my omp.json path segment 

```json
{
          "background": "#29315A",
          "foreground": "#3EC669",
          "leading_diamond": "\ue0b6",
          "properties": {
            "stack_count_enabled": true,
            "style": "full"
          },
          "style": "diamond",
          "template": "{{ if gt .StackCount 0 }}{{ .StackCount }}x{{ end }}\ue5ff {{ .Path }}",
          "trailing_diamond": "\ue0b4",
          "type": "path"
        }
```

Not that I think it matters as if I make a segment with simply .StackCount it never changes from 0

**Comment by JanDeDobbeleer** [comment]
It uses `(Get-Location -Stack).Count` to get that info, what's the output of that command in your case?

**Reply by JanDeDobbeleer** [reply]
@lewis-yeung something I missed when moving to the core module is this behavior:


```powershell
# Workaround for get-location/push-location/pop-location from within a module
# https://github.com/PowerShell/PowerShell/issues/12868
# https://github.com/JanDeDobbeleer/oh-my-posh2/issues/113
$global:OMP_GLOBAL_SESSIONSTATE = $PSCmdlet.SessionState

```

**Reply by QNetSo** [reply]
```PowerShell
> (Get-Location -Stack).Count
0
> pushd
>( Get-Location -Stack).Count
1
> pushd
> (Get-Location -Stack).Count
2
```

Prompt doesn't update :( - I removed the oh-my-posh prompt from the code above as on here it just showed as blocks. 

is there something I need to add to my powershell profile now?

**Reply by JanDeDobbeleer** [reply]
No, I need to make a fix for this. On my list for tomorrow!

**Reply by QNetSo** [reply]
I don't think .Writable works either - not to give you more work. And I don't use .Writable. but when I was trying to figure out the stackcount I tried the other path settings to see if I was missing something and .Writable would throw an error. Just FYI in case it's useful while your fixing the stack one ;)

**Reply by lewis-yeung** [reply]
> ```powershell
> # Workaround for get-location/push-location/pop-location from within a module
> # https://github.com/PowerShell/PowerShell/issues/12868
> # https://github.com/JanDeDobbeleer/oh-my-posh2/issues/113
> $global:OMP_GLOBAL_SESSIONSTATE = $PSCmdlet.SessionState
> ```

@JanDeDobbeleer I have tested this. The trick which we used before may not work now, while at least [this one](https://github.com/JanDeDobbeleer/oh-my-posh2/issues/113#issuecomment-490934765) is still useful. Similarly, we can make a simple function outside the `oh-my-posh-core` module and call it inside:

```powershell
function Get-PoshStackCount { (Get-Location -Stack).Count }
```

Or maybe you have a better solution that I haven't figured out. 😄

**Reply by JanDeDobbeleer** [reply]
Fix is live.
